.github/workflows/run-tests.yml

```diff
@@ -8,8 +8,44 @@ concurrency:
     group: ${{ github.workflow}}-${{ github.ref }}
     cancel-in-progress: true
 jobs:
+  check-formatting:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: 11
+          distribution: temurin
+          cache: maven
+
+      - name: Check formatting in core
+        run: mvn speedy-spotless:check --batch-mode --no-transfer-progress --update-snapshots -q -f ./dhis-2/pom.xml
+
+      - name: Check formatting in web
+        run:
+          mvn speedy-spotless:check --batch-mode --no-transfer-progress --update-snapshots -q -f ./dhis-2/dhis-web/pom.xml
+  analyze-dependencies:
+    runs-on: ubuntu-latest
+    if: "!contains(github.event.pull_request.labels.*.name, 'skip-check-dependencies')"
```

Comment:
Why would we want to skip this check? We don't allow skipping the formatting check as well 😋

Reply:
I am just being conservative for now.
I am not sure how this will work for the whole team and I want a way to make it not a blocker.
I think that after a while we can evaluate if remove this condition